### ISSUE TITLE
Question: Why not a type alias?

### ISSUE THREAD

**Issue by rodrigorc** [MathiasPius/fallible-option#1]
Hello! I was reading your [post](https://datavirke.dk/posts/fallible-missing-rust-error-handling/) and I was wondering: why not use a type alias instead? It will still communicate your intention better than the `Result<(), E>` and you will not be creating a new incompatible type for your errors.

I mean, something like this:
```
type Fallible<E> = std::result::Result<(), E>;
fn fail<E>(e: E) -> Fallible<E> {
    Err(e)
}
fn success<E>() -> Fallible<E> {
    Ok(())
}
```

Link to [playground ](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b6d4e4fc2c17628bc3d32d2696e43b13)with tests.

**Comment by MathiasPius** [comment]
It's a very good point, and it's a great way to get most of the way there.

One reason is that since type aliases are completely transparent, compiler errors which include these will show the aliased type, meaning if you fail to write correct code which calls or is itself a function that returns (a type-aliased) `Fallible<E>`, the signature of the function in question will say `Result<(), E>` in all the places you put `Fallible<E>`.

Some of the methods on `Result` also don't really make sense for the type alias. For instance, `map` acts on the successful case for `Result`, but it doesn't make sense at all to map a `Fallible<E>` *except* in the case where you want to map the error, which is what [map](https://docs.rs/fallible-option/latest/fallible_option/enum.Fallible.html#method.map) does.

tl;dr: Type alias gets you 80% there, if you are well *aware* that it is a type alias.

**Comment by xkr47** [comment]
> For instance, `map` acts on the successful case for `Result`, but it doesn't make sense at all to map a `Fallible<E>` _except_ in the case where you want to map the error, which is what [map](https://docs.rs/fallible-option/latest/fallible_option/enum.Fallible.html#method.map) does.

On the contrary one might still want to replace the "no value" with "some value" only for the case when the operation did not fail.. But yeah calling that `map` is perhaps a bit misleading..
